### DISCUSSION TITLE
What does the workspace folder do?

### DISCUSSION THREAD

**Discussion by YSCohen** [ScoopInstaller/Scoop#5307] [General]
The wiki [page](https://github.com/ScoopInstaller/Scoop/wiki/Scoop-Folder-Layout) on the folder layout is extremely incomplete.
mine has a few manifests in it

**Comment by niheaven** [comment]
It contains user-generated manifests or non-bucket manifests.

**Reply by YSCohen** [reply]
so it's a sort of custom local bucket?

**Reply by niheaven** [reply]
No, it's NOT a bucket, just a location to place customized manifests.

You could ignore it in most cases.

**Reply by YSCohen** [reply]
No, I understand that it's not a bucket. What I meant was that it's a place with manifests that you can install from. Is that the case, or what is it, exactly? What does it do? When are manifests put there? &c.

**Comment by HUMORCE** [comment]
> What I meant was that it's a place with manifests that you can install from. Is that the case, or what is it, exactly?

YES for Scoop and NO for users.

> What does it do?

The manifests in `workspaces` exists as dummy.

> When are manifests put there?

When Scoop generates an older version of the manifest, the auto-generated manifest will be stored there.

Example:

```
❯ scoop download python@3.11.0
INFO  Downloading 'python' (3.11.0) [64bit] from main bucket
WARN  Given version (3.11.0) does not match manifest (3.11.1)
WARN  Attempting to generate manifest for 'python' (3.11.0)
...
Writing updated python manifest
...

❯ ls ~\scoop\workspace

    Directory: C:\Users\humorce\scoop\workspace

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          12/26/2022  5:26 PM           4652 python.json
```

**Reply by YSCohen** [reply]
Aha. Thank you

**Comment by rashil2000** [comment]
As mentioned, the workspaces folder is for internal usage, and not meant for users.
